marconi-chain-index/src/Marconi/ChainIndex/Indexers/Utxo.hs

```diff
@@ -516,41 +529,49 @@ instance Buffered UtxoHandle where
     bracket_
         (SQL.execute_ c "BEGIN")
         (SQL.execute_ c "COMMIT")
-        (do
-         unless (null rows)
-             $ SQL.executeMany c
-                 [r|INSERT
-                    INTO transactions_outputs (
-                      address,
-                      txId,
-                      txIx,
-                      datum,
-                      datumHash,
-                      value,
-                      inlineScript,
-                      inlineScriptHash,
-                      slotNo,
-                      blockHash,
-                      spentSlotNo,
-                      spentBlockHash,
-                      spentTxId,
-                      txIndexInBlock
-                    ) VALUES
-                    (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)|] rows
-         unless (null spents)
-             $ SQL.executeMany c
-                   [r|UPDATE transactions_outputs
-                      SET spentSlotNo = ?,
-                          spentBlockHash = ?,
-                          spentTxId = ?
-                      WHERE txId = ? AND txIx = ?|] spents
-         )
+        (concurrently_
+         (unless
+          (null rows)
+          (SQL.executeMany c
+            [r|INSERT
+               INTO unspent_transactions (
+                 address,
+                 txId,
+                 txIx,
+                 datum,
+                 datumHash,
+                 value,
+                 inlineScript,
+                 inlineScriptHash,
+                 slotNo,
+                 blockHash,
+                 txIndexInBlock
+              ) VALUES
+              (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)|] rows))
+         (unless
+          (null spents)
+          (SQL.executeMany c
+           [r|INSERT
+              INTO spent (
+                txId,
+                txIx, slotNo, blockHash, spendTxId
+              ) VALUES
+              (?, ?, ?, ?, ?)|] spents)))
     -- We want to perform vacuum about once every 100
     when (toVacuume h) $ do
       rndCheck <- createSystemRandom >>= uniformR (1 :: Int, 100)
       when (rndCheck == 42) $ do
-        SQL.execute_ c [r|DELETE FROM transactions_outputs
-                          WHERE transactions_outputs.spentSlotNo IS NOT NULL|]
+        SQL.execute_ c [r|DELETE FROM
+                            unspent_transactions
+                          WHERE
+                            unspent_transactions.rowid IN (
+                              SELECT
+                                unspent_transactions.rowid
+                              FROM
+                                unspent_transactions
+                                JOIN spent ON unspent_transactions.txId = spent.txId
```

Comment:
when we left join the spent & unspent_transactions tables, why do we need to join them on the (txId, txIx)? rather than just txId